integreat_cms/cms/utils/external_calendar_utils.py

```diff
@@ -0,0 +1,263 @@
+        logger.debug(
+            "Event(event_id=%s, title=%s, start=%s, end=%s, content=%s...)",
+            event_id,
+            title,
+            start,
+            end,
+            content[:32],
+        )
+
+        return cls(
+            event_id=event_id,
+            title=title,
+            content=content,
+            start_date=start.date() if isinstance(start, datetime.datetime) else start,
+            start_time=start.time() if isinstance(start, datetime.datetime) else None,
+            end_date=end.date() if isinstance(end, datetime.datetime) else end,
+            end_time=end.time() if isinstance(end, datetime.datetime) else None,
+            is_all_day=not isinstance(start, datetime.datetime),
+            external_calendar_id=external_calendar_id,
+            categories=categories,
+        )
+
+    def to_event_form_data(self) -> dict:
+        """
+        Returns a dictionary of relevant data for the event form
+        :return: Dict of relevant data
+        """
+        return {
+            "start_date": self.start_date,
+            "start_time": self.start_time,
+            "end_date": self.end_date,
+            "end_time": self.end_time,
+            "is_all_day": self.is_all_day,
+            "has_not_location": True,
+            "external_calendar": self.external_calendar_id,
+            "external_event_id": self.event_id,
+        }
+
+    def to_event_translation_form_data(self) -> dict:
+        """
+        Returns a dictionary of relevant data for the event translation form
+        :return: Dict of relevant data
+        """
+        return {"title": self.title, "status": status.PUBLIC, "content": self.content}
+
+
+def import_events(calendar: ExternalCalendar, logger: logging.Logger) -> None:
+    """
+    Imports events from this calendar and sets or clears the errors field of the calendar
+
+    :param calendar: The external calendar
+    :param logger: The logger to use
+    """
+
+    errors: list[str] = []
+
+    _import_events(calendar, errors, logger)
+
+    if errors:
+        calendar.errors = "\n".join(errors)
+    else:
+        calendar.errors = ""
+
+    calendar.save()
+
+
+def _import_events(
+    calendar: ExternalCalendar, errors: list[str], logger: logging.Logger
+) -> None:
+    """
+    Imports events from this calendar and sets or clears the errors field of the calendar
+
+    :param calendar: The external calendar
+    :param logger: The logger to use
+    """
+    try:
+        ical = calendar.load_ical()
+    except IOError as e:
+        logger.error("Could not import events from %s: %s", calendar, e)
+        errors.append(_("Could not access the url of this external calendar"))
+        return
+    except ValueError as e:
+        logger.error("Malformed calendar %s: %s", calendar, e)
+        errors.append(
+            _("The data provided by the url of this external calendar are invalid")
```

Comment:
Seeing so many sands at the beach made me thirsty, I drank many water.
```suggestion
            _("The data provided by the url of this external calendar is invalid")
```